### PR TITLE
Update the README add RFC Stages

### PR DESCRIPTION
**Pull request by pzuraq**
Updates the process and templates to match [RFC 617](https://github.com/emberjs/rfcs/blob/master/text/0617-rfc-stages.md) (RFC Stages)

### CODE REVIEW

**Comment by kategengler** [comment]
Thanks! We shouldn't merge this until we are ready to switch to RFC Stages.

File: README.md
Comment (jenweber) [inline README.md:111]: Just for ease of community use/understanding, what do you think about moving the table above some of the other topics in the README?

**Review by jenweber — APPROVED** [review]
One thing that appears to be missing is the info about links to tracking things. I am not sure if that got lost in the shuffle/editing of the RFC itself, but for PRs that are merged, there should be some kind of tracking link that could go to a project board, Issue, PR, etc, just not necessarily the "RFCs Tracking" repo. The RFC has references to such a label that should be added as work progresses.

Otherwise looks great!

**Comment by kategengler** [comment]
@jenweber I don't think this is meant to be a comprehensive implementation of the RFC

**Review by wagenet — APPROVED** [review]

**Comment by wagenet** [comment]
I think this is a pretty crucial change since one major source of emo from people has been the claim that Ember is starting new features before properly wrapping up old ones. This would codify the process for that wrap-up and either make it clear that earlier features are wrapped up, or if they aren't, make it clear what work is necessary.

I'm starting a big push to grease the wheels of RFCs and PRs in general and I'm 100% on board with this.

File: README.md
Comment (wagenet) [inline README.md:19]: These links will need to be changed since PRs are no longer the best way to check things. Also merged PRs are not necessarily approved after Staging is implemented. We should probably point to rfcs.emberjs.com.

**Review by wagenet — CHANGES_REQUESTED** [review]
Looks good to me except the one issue I called out. We probably need to make sure to coordinate this with a blog post and we may also need rfcs.emberjs.com to be ready as well.

File: README.md
Comment (wagenet) [inline README.md:110]: Do we still want this? This number is going to grow a lot more quickly if we have PRs to update RFCs.

Comment (kategengler) [reply README.md:110]: Would we stop numbering RFCs then? 

It shouldn't be any more onerous for the numbers growing quickly, but it is annoying that it needs to be added after opening the PR. RFCs keep the # from their initial PR to 'Accepted'

Comment (wagenet) [reply README.md:110]: I don’t know whether we should stop numbering or not, but there’s an already incorrect impression that RFC-800 means that there were 800 RFCs that were at least proposed by that point. This only gets worse with more actual PRs now. We could just pick the number sequentially when we actually merge. Possibly could even do this with automation. Not a big deal either way and not a blocker.

File: README.md
Comment (kategengler) [reply README.md]: I addressed this in the latest commit. lmk if you think that is enough.

Comment (wagenet) [reply README.md]: Looks great.

**Review by wagenet — APPROVED** [review]

**Pull request merged**
